worker/src/test/java/gov/cms/ab2d/worker/service/JobProcessingServiceTest.java

```diff
@@ -0,0 +1,171 @@
+
+    @BeforeEach
+    void setUp() {
+        jobRepository.deleteAll();
+        userRepository.deleteAll();
+        sponsorRepository.deleteAll();
+
+        sponsor = createSponsor();
+        user = createUser(sponsor);
+        job = createJob(user);
+    }
+
+
+    @Test
+    @DisplayName("When a job is in submitted status, it can be put into progress upon starting processing")
+    void whenJobIsInSubmittedStatus_ThenJobShouldBePutInProgress() {
+        var processedJob = cut.putJobInProgress("S001");
+        assertThat(processedJob.getStatus(), is(JobStatus.IN_PROGRESS));
+    }
+
+    @Test
+    @DisplayName("When a job is not already in a submitted status, it cannot be put into progress")
+    void whenJobIsNotInSubmittedStatus_ThenJobShouldNotBePutInProgress() {
+        job.setStatus(JobStatus.IN_PROGRESS);
+        jobRepository.save(job);
+
+        var exceptionThrown = assertThrows(IllegalArgumentException.class,
+                () -> cut.putJobInProgress("S001"));
+
+        assertThat(exceptionThrown.getMessage(), is("Job S001 is not in SUBMITTED status."));
+    }
+
+
+    @Test
+    @DisplayName("When a job is in submitted status, it can be processed")
+    void processJob() throws IOException {
+
+        job.setStatus(JobStatus.IN_PROGRESS);
+        jobRepository.save(job);
+        var contract = createContract(sponsor);
+
+        var processedJob = cut.processJob("S001");
+
+        assertThat(processedJob.getStatus(), is(JobStatus.SUCCESSFUL));
+        assertThat(processedJob.getStatusMessage(), is("100%"));
+        assertThat(processedJob.getExpiresAt(), notNullValue());
+    }
+
+    @Test
+    @DisplayName("When a job is in submitted by the parent user, it process the contracts for the children")
+    void whenJobSubmittedByParentUser_ProcessAllContractsForChildrenSponsors() throws IOException {
+
+        // create parent sponsor
+        final Sponsor parentSponsor = createSponsor();
+        parentSponsor.setOrgName(parentSponsor.getOrgName() + " - PARENT");
+        parentSponsor.setLegalName(parentSponsor.getLegalName() + " - PARENT");
+
+        // associate the parent to the child
+        final Sponsor childSponsor = user.getSponsor();
+        childSponsor.setParent(parentSponsor);
+        sponsorRepository.save(childSponsor);
+
+        // switch the user to the parent sponsor
+        user.setSponsor(parentSponsor);
+        userRepository.save(user);
+
+
+        job.setStatus(JobStatus.IN_PROGRESS);
+        jobRepository.save(job);
+        var contract = createContract(this.sponsor);
+
+        var processedJob = cut.processJob("S001");
+
+        assertThat(processedJob.getStatus(), is(JobStatus.SUCCESSFUL));
+        assertThat(processedJob.getStatusMessage(), is("100%"));
+        assertThat(processedJob.getExpiresAt(), notNullValue());
+    }
+
+    private Sponsor createSponsor() {
+        Sponsor sponsor = new Sponsor();
+        sponsor.setId(Long.valueOf(random.nextInt()));
```

Comment:
Why do we need to generate a primary key? Hibernate's sequence will take care of that.

Reply:
done